### DISCUSSION TITLE
Preload or Load? Which one?

### DISCUSSION THREAD

**Discussion by j-dominguezp** [adonisjs/core#2545] [Help] [answered]
I'm getting this deprecation warning: 

> "Model.preload()" is deprecated. Use "Model.load()" instead: DeprecationWarning

Preload seems to be well documented, so I'm kind of confused.

If there is a difference, then any help understanding it would be awesome.

Well, any help or information will be greatly appreciated.

Kudos.

**Comment by vookio** [comment, marked as answer]
mentioned at the end of this page: 
[https://docs.adonisjs.com/releases/out-of-preview](https://docs.adonisjs.com/releases/out-of-preview)
```
- deprecate model.preload in favor of model.load be4e0d2f6b1a8
```

**Reply by j-dominguezp** [reply]
Thanks for your response, I've already seen that, I was looking more on an explanation on what's the difference between the non-deprecated (and documented) one and the deprecated one.

Anyway, thanks for your fast answer as this might be helpful to other people in the community. 😀

**Comment by thetutlage** [comment]
They are functionally the same, however the naming is used to suit the phase in which you are loading the relationships.

- **preload** means, load the relationship along side the main results. So preload is used with the query builder instance. `Post.query().preload('comments')`
- **load** means, load the relationships after you have the parent model instance. `load` is used with the model instance. `post.load('comments')`

**Reply by j-dominguezp** [reply]
Thanks, this was exactly what I was looking for! Congrats on v5 release!
